modules/peridynamics/src/postprocessors/BondStatusConvergedPostprocessorPD.C

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BondStatusConvergedPostprocessorPD.h"
+
+registerMooseObject("PeridynamicsApp", BondStatusConvergedPostprocessorPD);
+
+InputParameters
+BondStatusConvergedPostprocessorPD::validParams()
+{
+  InputParameters params = ElementPostprocessor::validParams();
+  params.addClassDescription(
+      "Postprocessor to check whether the bond status is converged within a time step");
+
+  return params;
+}
+
+BondStatusConvergedPostprocessorPD::BondStatusConvergedPostprocessorPD(
+    const InputParameters & parameters)
+  : ElementPostprocessor(parameters),
+    _bond_status_var(&_subproblem.getStandardVariable(_tid, "bond_status")),
+    _pdmesh(dynamic_cast<PeridynamicsMesh &>(_mesh)),
+    _dim(_pdmesh.dimension())
+{
+}
+
+void
+BondStatusConvergedPostprocessorPD::initialize()
+{
+  _bond_status_updated_times = 0;
+}
+
+void
+BondStatusConvergedPostprocessorPD::execute()
+{
+  if ((_bond_status_var->getElementalValueOld(_current_elem) -
```

Comment:
Is the old value the value from the previous time step, or the value from the previous Picard iteration?

Reply:
You are right that this should be the value from previous Picard iteration, since this is only activated during Picard iteration. But for the first Picard iteration,  the values are from previous time step.

Reply:
The aux kernel for bond status will be executed at the end of the each time step as well as each Picard iteration.